symfony/plugins/orangehrmTimePlugin/Report/ProjectReportData.php

```diff
@@ -0,0 +1,105 @@
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Report;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Report\ReportData;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Traits\Service\NormalizerServiceTrait;
+use OrangeHRM\Core\Traits\Service\NumberHelperTrait;
+use OrangeHRM\Time\Api\Model\ProjectModel;
+use OrangeHRM\Time\Dto\ProjectReportSearchFilterParams;
+use OrangeHRM\Time\Traits\Service\ProjectServiceTrait;
+
+class ProjectReportData implements ReportData
+{
+    use ProjectServiceTrait;
+    use NumberHelperTrait;
+    use DateTimeHelperTrait;
+    use NormalizerServiceTrait;
+
+    private ProjectReportSearchFilterParams $filterParams;
+
+    public function __construct(ProjectReportSearchFilterParams $filterParams)
+    {
+        $this->filterParams = $filterParams;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function normalize(): array
+    {
+        $projects = $this->getProjectService()->getProjectDao()->getProjectReportCriteriaList($this->filterParams);
+        $fromDateYmd = $this->getDateTimeHelper()->formatDateTimeToYmd($this->filterParams->getFromDate());
+        $toDateYmd = $this->getDateTimeHelper()->formatDateTimeToYmd($this->filterParams->getToDate());
+        $projectId = $this->filterParams->getProjectId();
+        $includeTimesheet = $this->filterParams->getIncludeApproveTimesheet() === null ?
+            'all' : $this->filterParams->getIncludeApproveTimesheet();
+        $projectActivityDetailsReportURL = '/time/displayProjectActivityDetailsReport';
+        $result = [];
+        foreach ($projects as $project) {
+            $activityId = $project['activityId'];
+            $result[] = [
+                ProjectReport::PARAMETER_ACTIVITY_ID => $activityId,
+                ProjectReport::PARAMETER_ACTIVITY_NAME => $project['name'],
+                ProjectReport::PARAMETER_ACTIVITY_DELETED => $project['deleted'],
+                ProjectReport::PARAMETER_TIME => $this->getNumberHelper()
+                    ->numberFormat((int)$project['totalDuration'] / 3600, 2),
```

Comment:
Seems this type casting is wrong, please cover this in unit tests
```suggestion
                    ->numberFormat((float)$project['totalDuration'] / 3600, 2),
```

Reply:
done